src/environmentd/src/http.rs

```diff
@@ -408,7 +411,7 @@ impl InternalHttpServer {
                 "/api/catalog/check",
                 routing::get(catalog::handle_catalog_check),
             )
-            .layer(Extension(AuthedUser(SYSTEM_USER.clone())))
+            .layer(middleware::from_fn(internal_http_auth))
```

Comment:
can you add a test for setting the header to mz_support, mz_system, and some other random string?

Reply:
Yeah, I looked briefly at the tests and was pretty lost. Can you point me in the right direction as to where I would add these cases? I didn't see any tests for this internal http endpoint.

Reply:
Hm, if it were the normal endpoint then `test_http_sql` in `environmentd/tests/server.rs` will run the tests in testdata/http/; adding your test to the test/http/post file would be enough.

In the test_leader_promotion tests in server.rs we hit the internal http endpoint and assert on the result. That's probably the approach I'd take. Another idea is to add support to test_http_sql to hit the internal endpoint (like today we can test post requests and ws requests)